### PR TITLE
merge cgiChunk into master

### PR DESCRIPTION
**Pull request by tblaase**
tester is now running every test, but still needs to be improved regarding the performance and properly disconnecting clients

### CODE REVIEW

File: server/config/42tester.conf [hunk 1]
```diff
@@ -36,12 +40,17 @@ server {
...
 # /post_body must answer anything to POST request with a maxBody of 100
 
-	client_max_body_size 100; MB?
-	client_body_buffer_size 100; MB?
+	client_max_body_size 100; bytes
+	client_body_buffer_size 100; bytes
 
 	location /directory/ {
-		method GET
+		method GET POST
```
Comment (aenglert42) [inline server/config/42tester.conf:47]: remove POST from allowed methods

File: server/config/42tester.conf [hunk 2]
```diff
@@ -36,12 +40,17 @@ server {
...
 		root /42tester/YoupiBanane/
 		index_page youpi.bad_extension
 	}
+
+	# location /uploads/ {
```
Comment (aenglert42) [inline server/config/42tester.conf:52]: remove comment

File: server/src/CgiResponse.cpp [hunk 3]
```diff
@@ -79,28 +65,51 @@ bool CgiResponse::checkForMandatoryHeaders(string& headerLine)
...
+	return (_body);
+}
+
 void CgiResponse::sendResponse()
 {
-	string headers = this->constructHeader();
-	send(_cgiFD, headers.c_str(), headers.length(), 0);
-	tunnelResponse(_cgiFD, _clientFD);
+	cout << "sending response" << " body" << this->_body << endl;
+	string headers  = "HTTP/1.1 200 OK\r\nContent-Length: " + std::to_string(this->_body.length()) + "\r\n\r\n"; // @Tam please do not send a hardcoded response, what if the cgi crahed or something went wrong?
```
Comment (aenglert42) [inline server/src/CgiResponse.cpp:88]: is this still needed?

File: server/src/Server.cpp [hunk 4]
```diff
@@ -446,30 +472,40 @@ void Server::handleRequest(int clientFd) // i is the index from the evList of th
...
 				this->_socketHandler->addKeepAlive(clientFd);
 
-			if (isCgi == true)
+			/* if (isCgi == true)
 			{
 				this->applyCurrentConfig(request);
 				cgi_handle(request, clientFd, this->_currentConfig);
 			}
-			else if (request.getMethod() == "POST" || request.getMethod() == "PUT")
+			else  */if (request.getMethod() == "POST" || request.getMethod() == "PUT")
```
Comment (aenglert42) [inline server/src/Server.cpp:492]: check if only enabled for tester

**Review by aenglert42 — CHANGES_REQUESTED** [review]
pls review my comments

File: server/src/CgiResponse.cpp (hunk 3 above)
Comment (tblaase) [reply server/src/CgiResponse.cpp:88]: fixed with dc9cc883d2cfd93bc5d7bd35739f1eba3d37851c

File: server/config/42tester.conf (hunk 2 above)
Comment (tblaase) [reply server/config/42tester.conf]: dc9cc883d2cfd93bc5d7bd35739f1eba3d37851c fixed it

File: server/src/CgiResponse.cpp (hunk 3 above)
Comment (tblaase) [reply server/src/CgiResponse.cpp:88]: fixed with a84262434613201308be1f5a9f8889051bb2be26

File: server/src/Server.cpp (hunk 4 above)
Comment (tblaase) [reply server/src/Server.cpp]: 9ecbfbcd4f8b6eca5c98ebb9a119f2d6a6b4ae08 fixed it

**Review by tblaase — COMMENTED** [review]
fixed all your stuff, pls confirm

**Pull request merged**